src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -127,13 +127,27 @@ bool emberAfGeneralCommissioningClusterArmFailSafeCallback(app::CommandHandler *
                                                            const app::ConcreteCommandPath & commandPath,
                                                            const Commands::ArmFailSafe::DecodableType & commandData)
 {
-    DeviceControlServer * server = &DeviceLayer::DeviceControlServer::DeviceControlSvr();
-    CheckSuccess(server->ArmFailSafe(System::Clock::Seconds16(commandData.expiryLengthSeconds)), Failure);
+    DeviceControlServer * server      = &DeviceLayer::DeviceControlServer::DeviceControlSvr();
+    FailSafeContext & failSafeContext = server->GetFailSafeContext();
 
-    Commands::ArmFailSafeResponse::Type response;
-    response.errorCode = CommissioningError::kOk;
-    response.debugText = CharSpan("", 0);
-    CheckSuccess(commandObj->AddResponseData(commandPath, response), Failure);
+    /*
+     * If the fail-safe timer was not currently armed, the the fail-safe timer SHALL be armed.
+     * If the fail-safe timer was currently armed, and current accessing fabric matches the fail-safe
+     * context’s Fabric Index, then the fail-safe timer SHALL be re-armed.
+     */
+    if (!failSafeContext.IsFailSafeArmed() ||
+        failSafeContext.MatchesFailSafeContextFabricIndex(commandObj->GetAccessingFabricIndex()))
+    {
+        Commands::ArmFailSafeResponse::Type response;
+
+        CheckSuccess(server->ArmFailSafe(System::Clock::Seconds16(commandData.expiryLengthSeconds)), Failure);
+        response.errorCode = CommissioningError::kOk;
+        CheckSuccess(commandObj->AddResponseData(commandPath, response), Failure);
+    }
+    else
+    {
+        LogErrorOnFailure(commandObj->AddStatus(commandPath, Protocols::InteractionModel::Status::Busy));
```

Comment:
See https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/4956 for the behavior we should have here.

Reply:
Updated